### ISSUE TITLE
Disable dnd for single item to column

### ISSUE THREAD

**Issue by DuckyDisciple** [isaacHagoel/svelte-dnd-action#489]
I'm not sure if this is a feature to add or if there just isn't an example of this out there. I'm looking to prevent a specific item from being moved into a specific dropzone based on a condition. For example, I have a project that has unmet requirements to change statuses. I want to be able to move that item backwards through the workflow, but if the requirement is not met, I want to lock it down from being moved forward through the statuses.

The way I currently handle this is when the item is dropped, it makes a call to the backend where it checks the requirements and if it fails, reverts the status. I also have code running on the frontend to update the view as the statuses change. This causes the project to move, then a second later jump back to the previous column, and the order changes as well.

Another idea I had is to lock down the drop from other functionality when the item with unmet requirements starts to move, but that seems like a very buggy way to do things to make sure the column gets re-enabled again.

**Comment by isaacHagoel** [comment]
The two ways that come to mind are:
1. Use two types: "allowed" and "not allowed" and assign them to dnd zones dynamically when more zones become allowed.
2. As you offered, disable drop from others on the zones that are not allowed. If you tried it and encountered bugs please let me know. It's supposed to work well (if you have nested zones I recommend using the latest version of the lib).
In both cases I would use a store to orchestrate the whole thing, and in my mind the code looks quite simple.
If you want you can create a simple REPL and I can help you get it over the finish line.

**Comment by DuckyDisciple** [comment]
> 2. As you offered, disable drop from others on the zones that are not allowed. If you tried it and encountered bugs please let me know. It's supposed to work well (if you have nested zones I recommend using the latest version of the lib).

I started to go this route and I am very close with it, but there is a problem. I have a store that holds locked columns and that part works great. I set up functions to lock columns based on factors in the selected item, which also works great. The only part I'm stuck on right now is the event handling. I tried adding `on:drag` to the item, but realized that this is technically not a draggable element which is why that isn't triggering. I added `on:mousedown` and it worked, however it removed the default functionality and wouldn't allow me to drag items between boards.

Is there a way to loop some sort of trigger in so that when a card is dragged or the mousedown event occurs that it can send a callback to a function that gets passed in?
